Sources/Utility/Process.swift

```diff
@@ -0,0 +1,295 @@
+        posix_spawnattr_setpgroup(&attributes, 0)
+
+        // Set the attribute flags.
+        var flags = POSIX_SPAWN_SETSIGMASK | POSIX_SPAWN_SETSIGDEF
+        flags |= POSIX_SPAWN_SETPGROUP
+
+        // Close all other files by default.
+        //
+        // FIXME: Note that this is an Apple-specific extension, and we will have to
+        // do something else on other platforms (and unfortunately, there isn't
+        // really an easy answer other than using a stub executable).
+      #if os(macOS)
+        flags |= POSIX_SPAWN_CLOEXEC_DEFAULT
+      #endif
+
+        posix_spawnattr_setflags(&attributes, Int16(flags))
+
+        // Setup the file actions.
+      #if os(macOS)
+        var fileActions: posix_spawn_file_actions_t? = nil
+      #else
+        var fileActions = posix_spawn_file_actions_t()
+      #endif
+        posix_spawn_file_actions_init(&fileActions)
+
+        // Workaround for https://sourceware.org/git/gitweb.cgi?p=glibc.git;h=89e435f3559c53084498e9baad22172b64429362
+        let devNull = strdup("/dev/null")
+        defer { free(devNull) }
+        // Open /dev/null as stdin.
+        posix_spawn_file_actions_addopen(&fileActions, 0, devNull, O_RDONLY, 0)
+
+        var rv = libc.pipe(&saveOutputPipe)
+        guard rv == 0 else {
+            throw SystemError.pipe(rv)
+        }
+
+        if outputStream != nil {
+            // Open the write end of the pipe as stdout and stderr, if desired.
+            posix_spawn_file_actions_adddup2(&fileActions, saveOutputPipe[1], 1)
+            posix_spawn_file_actions_adddup2(&fileActions, saveOutputPipe[1], 2)
+
+            // Close the other ends of the pipe.
+            posix_spawn_file_actions_addclose(&fileActions, saveOutputPipe[0])
+            posix_spawn_file_actions_addclose(&fileActions, saveOutputPipe[1])
+        } else {
+            posix_spawn_file_actions_adddup2(&fileActions, 1, 1)
+            posix_spawn_file_actions_adddup2(&fileActions, 2, 2)
+        }
+
+        let argv: [UnsafeMutablePointer<CChar>?] = arguments.map{ $0.withCString(strdup) }
+        defer { for case let arg? in argv { free(arg) } }
+        
+        let env: [UnsafeMutablePointer<CChar>?] = environment.map{ "\($0.0)=\($0.1)".withCString(strdup) }
+        defer { for case let arg? in env { free(arg) } }
+
+        rv = posix_spawnp(&processID, argv[0], &fileActions, &attributes, argv + [nil], env + [nil])
+
+        posix_spawn_file_actions_destroy(&fileActions)
+        posix_spawnattr_destroy(&attributes)
+
+        guard rv == 0 else {
+            throw SystemError.posix_spawn(rv, arguments)
+        }
+    }
+
+    /// Blocks the calling process until the subprocess finishes execution.
+    ///
+    /// - Throws: Process.Error
+    public func waitUntilExit() throws {
+        assert(launched, "The process is not yet launched.")
+        // FIXME: We should probably read the output from somewhere else, maybe in a seperate thread?
```

Comment:
Yes, this has to be done.

As currently written the launched process may end up blocked waiting for its output pipe to be read from, so it is critical that we be consuming the output in the background.